crates/holochain/src/conductor/cell.rs

```diff
@@ -824,6 +824,15 @@ impl Cell {
                             |row| row.get(0),
                         )?;
 
+                        if receipt_count >= required_validation_count as usize {
+                            // If we have enough receipts then set receipts to complete.
+                            //
+                            // Don't fail here if this doesn't work, it's only informational. Getting
+                            // the same flag set in the authored db is what will stop the publish
+                            // workflow from republishing this op.
+                            set_receipts_complete(txn, &receipt_op_hash, true).ok();
+                        }
```

Comment:
Not thrilled about this. Publishing is done from the authored database, receipts are received into the dht database, but then receipts complete is separately placed in the authored database so that publish can see it. I'm now putting it in both.

Reasoning: I have to look up validation receipts from the dht database, but this field would always be `false` unless I went to look it up from the authored database. It seems cleaner to set this flag in both than to have to read from dht and authored to get validation receipts.

Reply:
I was wondering if we even want to include receipts_complete in the response. The app already knows how many receipts are required for a given EntryType, and it can find the length of the receipts vec itself. That field is actually a duplication of state, useful in Holochain queries, but I don't know that we need to extend it to the user. And then we wouldn't need this less-than-thrilling update, right?

Reply:
Yes the app can figure this out, assuming it knows how to find the constant for the default number of required receipts and figures out how to retrieve the number required by the app if that is set. So not setting this here is definitely an option. It does seem convenient to have that pre-computed and available to the user, given that we are computing it already. Yes we have to copy it, but we're not taking up any extra space to copy it because we're already storing `false` in that database column.